packages/provider/src/RelaySelectionManager.ts

```diff
@@ -76,8 +76,8 @@ export class RelaySelectionManager {
             relayInfo
           }
         } else {
-          // TODO: do not throw! The preferred relay may be removed since.
-          throw new Error('Could not find register event for the winning preferred relay')
+          this.logger.error('Could not find registration info in the RelayRegistrar for the selected preferred relay')
+          return undefined
```

Comment:
missing a test to show the outcome of this `return undefined`

Reply:
> missing a test to show the outcome of this `return undefined`

It is a bit hard to observe anything happening here other than adding a `sinon.spy` to the logger and then check the error is logged. This code is explicitly about ignoring an error.
Do you want a spy test or do you have a better idea?

Reply:
The method didn't return `undefined` before, and now it does.
I'm not sure how it behaves, and thus I think it should be tested somehow.

You can do a manual test, e.g. by doing `yarn link` into ctf and test it.

Reply:
> The method didn't return `undefined` before, and now it does. I'm not sure how it behaves, and thus I think it should be tested somehow.
> 
> You can do a manual test, e.g. by doing `yarn link` into ctf and test it.

This method is, however, explicitly marked as `Promise<RelayInfo | undefined>`, with `undefined` meaning "no more [preferred] relays", as handled in `selectNextRelay`